pkg/utils/event.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package utils
+
+import "sync"
+
+// Event is a synchronization primitive that lets an arbitrary number
+// of concurrent waiters to wait for an event to happen 1 or more times.
+type Event struct {
```

Comment:
```
type Event struct {
	mu     sync.Mutex
	unlock sync.Once
}

func NewEvent() {
	event := &Event{}
	event.mu.Lock()
	return event
}

func (event *Event) Wait() {
	event.mu.Lock()
	event.mu.Unlock()
}

func (event *Event) Fire() {
	event.unlock.Do(func(){
		event.mu.Unlock()
	})
}
```

```
// simplest but only allows close being called once
type Event struct {
	ch chan struct{}
}

func NewEvent() {
	return &Event{make(chan struct{})}
}

func (event *Event) Wait() { <-event.ch }
func (event *Event) Close() { close(event.ch) }
```

Reply:
yep, considered both of those options but the benefit of being able to use it like a sync.Mutex (zero value is initialized and ready to go) outweighed those two implementations for me. nonetheless, with the removal of the `noListener` struct, this also is unnecessary, and i removed it entirely

Reply:
Sure, makes sense, just giving alternatives here that are bit easier to follow.